### ISSUE TITLE
App needs to be restarted to see the new displayname set by other device

### ISSUE THREAD

**Issue by iequidoo** [deltachat/deltachat-desktop#3624]
- Operating System (Linux/Mac/Windows/iOS/Android): Debian GNU/Linux 12
- Delta Chat Version: 1.42.2 (git: v1.33.0-555-g9da03dd2)
- Expected behavior: The new displayname ("Your Name") is displayed w/o an app restart
- Actual behavior: Need to restart the app
- Steps to reproduce the problem:
  - Change the displayname aka "Your Name" on other device
  - Check if the new value is seen in the app

Notes: in DC Android it works as expected. In DC cli, too. Do we have any cache in Desktop?

**Comment by Simon-Laux** [comment]
we have a cache/store for some settings in desktop, is there an event when it changes?

the cache is there to get core values faster, because it is sync, while the core api is async, though we could probably change the code to have less places that rely on it with the cost that we may have to add some loading delays.

**Comment by iequidoo** [comment]
Unfortunately, no special event is emitted for this. Only `IncomingMsgBunch` is emitted for sync messages, can it be used to update the displayname and other things (e.g. the avatar) that can be synchronised?

**Comment by Simon-Laux** [comment]
so @farooqkz listen for `IncomingMsgBunch` in settings store and reload the whole message store on that event, don't forget to add a comment that explains that there is no dedicated event for this yet and maybe link to this issue.

**Comment by iequidoo** [comment]
Yes, relying on `IncomingMsgBunch` isn't a good idea, because it only means that some message is received, but it might not be yet applied.

**Comment by r10s** [comment]
> @r10s would it be useful in mobile apps?

no. we do not cache config values, this is not needed, things are fast enough. iirc, not caching is no option as all calls are async on drsktop? or did that change meanwhile?

**Comment by iequidoo** [comment]
So, i'm going to add some event like
```
ConfigChanged {
    key: Config,
    val: Option<String>,
}
```
It will also replace `SelfavatarChanged`.

**Comment by Simon-Laux** [comment]
> EDIT: How about ConfigSynced (with the same params)?

I agree, that would be nice, though value is probably not needed, we can always fetch the newest version of it with a core call.

**Comment by iequidoo** [comment]
> I agree, that would be nice, though value is probably not needed, we can always fetch the newest version of it with a core call.

Decided not to add a value, otherwise it would be logged which might not be good for privacy when sharing logs

**Comment by iequidoo** [comment]
Merged the core PR

**Comment by iequidoo** [comment]
Merged another core PR fixing the previous one, now should work
